crates/project/src/dap_store.rs

```diff
@@ -1164,38 +1156,42 @@ impl DapStore {
         evaluate_name: Option<String>,
         cx: &mut ModelContext<Self>,
     ) -> Task<Result<()>> {
-        let Some(client) = self.client_by_id(client_id) else {
-            return Task::ready(Err(anyhow!("Could not find client: {:?}", client_id)));
-        };
-
         let supports_set_expression = self
             .capabilities_by_id(client_id)
             .supports_set_expression
             .unwrap_or_default();
 
-        cx.background_executor().spawn(async move {
-            if let Some(evaluate_name) = supports_set_expression.then(|| evaluate_name).flatten() {
-                client
-                    .request::<SetExpression>(SetExpressionArguments {
-                        expression: evaluate_name,
-                        value,
-                        frame_id: Some(stack_frame_id),
-                        format: None,
-                    })
-                    .await?;
-            } else {
-                client
-                    .request::<SetVariable>(SetVariableArguments {
-                        variables_reference,
-                        name,
-                        value,
-                        format: None,
-                    })
-                    .await?;
-            }
+        if let Some(evaluate_name) = supports_set_expression.then(|| evaluate_name).flatten() {
+            let arguments = SetExpressionArguments {
+                expression: evaluate_name,
+                value,
+                frame_id: Some(stack_frame_id),
+                format: None,
+            };
 
-            Ok(())
-        })
+            let task = self.request_dap::<SetExpression>(client_id, arguments, cx);
+
+            cx.background_executor().spawn(async move {
+                task.await?;
+
+                Ok(())
+            })
+        } else {
+            let arguments = SetVariableArguments {
+                variables_reference,
+                name,
+                value,
+                format: None,
+            };
+
+            let task = self.request_dap::<SetVariable>(client_id, arguments, cx);
+
+            cx.background_executor().spawn(async move {
```

Comment:
I think this wrapping it inside a background task is not needed, as you already do this inside the request_dap method.

Reply:
I tried doing 

```rust
task.map(|res| res.map(|response| response.variables)) 
```

before but the return type missed with the function. I do think you're right that we shouldn't do this, I'm going to change the function's return type instead. Thanks for the feedback!!